HeroFleeRules.ai

```diff
@@ -31,7 +31,7 @@
   call ActionRule(not race_has_moonwells or TownCountDone(racial_farm) <= 0, ACTION_MOONWELLS,-1000000)
   if nearest_neutral[hFountainID] != null then
     call ActionRule(DistanceBetweenPoints_dk(GetUnitLoc(nearest_neutral[hFountainID]),l) > DistanceBetweenPoints(home_location,l), ACTION_MOONWELLS, 18) // Use moonwells if they closer than the fountain
-  else
+  elseif race_has_moonwells then
```

Comment:
Not required action rule line 31 above disables moonwells already.

Reply:
Regardless of whether it is disabled or not, the ActionRule will still be written here. If this layer of judgment is not added, it will be an unnecessary judgment for races that do not use ymounwells. 
Therefore, I have tried to fundamentally reduce invalid rules
At least ... I hope so

Reply:
It will still be -1000000-18 = -99,982 total priority for ACTION_MOONWELLS so its still impossible.

Reply:
but It seems that `action_points` will still stack, I hope it reduces the computational load here